src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParser.cs

```diff
@@ -1511,13 +1490,19 @@ private int ScanDecimal()
         private char ScanHex(int c)
         {
             int i = 0;
-            int d;
 
             if (_pos + c <= _pattern.Length)
             {
-                for (; c > 0 && ((d = HexDigit(_pattern[_pos++])) >= 0); c -= 1)
+                for (; c > 0; c -= 1)
                 {
-                    i = (i * 0x10) + d;
+                    int d;
+                    char ch = _pattern[_pos++];
```

Comment:
Can this use our HexConverter internal utility instead?

Reply:
I looked and it seemed a lot more indirection that surely is more expensive than this is given we're just converting two digits. But I can look again if you like.

Reply:
> I looked and it seemed a lot more indirection that surely is more expensive than this is given we're just converting two digits

What indirection are you referring to?  I'm referring to https://github.com/dotnet/runtime/blob/b898a540f75d78ca55ee33288a4fd62b1ab69fa2/src/libraries/Common/src/System/HexConverter.cs#L357-L361. It uses a lookup table.  If it were more expensive then the multiple checks being performed here, then that centralized implementation would be changed to the more efficient thing.